homeassistant/components/flipr/__init__.py

```diff
@@ -1,22 +1,47 @@
 """The Flipr integration."""
 
+from collections import Counter
+import logging
+
+from flipr_api import FliprAPIRestClient
+
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import Platform
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
 
-from .const import DOMAIN
+from .const import CONF_ENTRY_FLIPR_COORDINATORS, DOMAIN
 from .coordinator import FliprDataUpdateCoordinator
 
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.SENSOR]
 
+_LOGGER = logging.getLogger(__name__)
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up Flipr from a config entry."""
+    """Set up flipr from a config entry."""
     hass.data.setdefault(DOMAIN, {})
 
-    coordinator = FliprDataUpdateCoordinator(hass, entry)
-    await coordinator.async_config_entry_first_refresh()
-    hass.data[DOMAIN][entry.entry_id] = coordinator
+    # Detect invalid old config entry and raise error if found
+    detect_invalid_old_configuration(hass, entry)
+
+    config = entry.data
+
+    username = config[CONF_EMAIL]
+    password = config[CONF_PASSWORD]
+
+    _LOGGER.debug("Initializing Flipr client %s", username)
+    client = FliprAPIRestClient(username, password)
+    ids = await hass.async_add_executor_job(client.search_all_ids)
+
+    _LOGGER.debug("List of devices ids : %s", ids)
+
+    flipr_coordinators = []
+    for flipr_id in ids["flipr"]:
+        flipr_coordinator = FliprDataUpdateCoordinator(hass, entry, flipr_id)
+        await flipr_coordinator.async_config_entry_first_refresh()
+        flipr_coordinators.append(flipr_coordinator)
+    hass.data[DOMAIN][CONF_ENTRY_FLIPR_COORDINATORS] = flipr_coordinators
```

Comment:
If you're only storing coordinators, just store it in `hass.data[DOMAIN[entry.entry_id]`

Reply:
In the future, please migrate to use `entry.runtime_data`

Reply:
OK, implemented it with entry.runtime_data directly.